### PR TITLE
Plugins: Add notices to docs to prevent NPX commands from hanging

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR adds a notice box below shell code examples that mention `npx @grafana/toolkit` with a workaround for peer dependency issues with NPM 7+ that cause the commands to hang in terminal.

**Which issue(s) this PR fixes**:

Related #34807

**Special notes for your reviewer**:

### CODE REVIEW

**Review by wbrowne — APPROVED** [review]

File: docs/sources/developers/plugins/_index.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ Open the terminal, and run the following command in your [plugin directory]({{<
 npx @grafana/toolkit plugin:create my-grafana-plugin
 ```
 
+> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. If so use `npx --legacy-peer-deps <command to run>`.
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/_index.md:20]:
```suggestion
> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. The workaround is to use `npx --legacy-peer-deps <command to run>`.
```

File: docs/sources/developers/plugins/sign-a-plugin.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ Public plugins need to be reviewed by the Grafana team before you can sign them.
    npx @grafana/toolkit plugin:sign
    ```
 
+> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. If so use `npx --legacy-peer-deps <command to run>`.
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/sign-a-plugin.md:44]:
```suggestion
> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. The workaround is to use `npx --legacy-peer-deps <command to run>`.
```
```

File: packages/grafana-toolkit/README.md [hunk 3]
```diff
@@ -15,6 +15,8 @@ yarn install
 yarn dev
 ```
 
+> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. If so use `npx --legacy-peer-deps <command to run>`.
```
Comment (achatterjee-grafana) [inline packages/grafana-toolkit/README.md:18]:
```suggestion
> **Note:** If running NPM 7+ the `npx` commands mentioned in this article may hang. The workaround is to use `npx --legacy-peer-deps <command to run>`.
```
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Review by mckn — APPROVED** [review]
LGTM! Awesome work.

**Pull request merged**